### PR TITLE
Implement order trash and untrash handling for Dokan

### PR DESCRIPTION
**Pull request by mralaminahamed**
### All Submissions:
* [x] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [x] My code satisfies feature requirements
* [ ] My code is tested
* [ ] My code passes the PHPCS tests
* [x] My code has proper inline documentation
* [ ] I've included related pull request(s) (optional)
* [ ] I've included developer documentation (optional)
* [ ] I've added proper labels to this pull request

### Changes proposed in this Pull Request:
This PR updates the OrderEventListener class to properly handle order trashing and untrashing in accordance with WooCommerce's HPOS feature. It ensures that when an order is trashed or untrashed, the status is correctly updated in both WooCommerce and Dokan-specific tables, maintaining consistency across the systems.

### Related Pull Request(s)
* N/A

### Closes 
* Closes https://github.com/getdokan/client-issue/issues/157

### How to test the changes in this Pull Request:
1. Enable HPOS in WooCommerce settings.
2. Create a test order.
3. Trash the order and verify that the status is updated correctly in both WooCommerce and Dokan tables.
4. Untrash the order and verify that the status is restored correctly in both WooCommerce and Dokan tables.
5. Check the error logs to ensure proper logging of trash/untrash actions.

### Changelog entry
*Implement order trash and untrash handling for Dokan*

This PR updates the OrderEventListener to properly handle order trashing and untrashing. It ensures that order statuses are correctly synchronized between WooCommerce and Dokan-specific tables, implements WooCommerce's trash metadata handling, and adds error logging for improved debugging.

### Before Changes
Previously, the OrderEventListener did not properly handle order trashing and untrashing when HPOS was enabled, potentially leading to inconsistencies between WooCommerce and Dokan order statuses.

### After Changes
With these changes, order trashing and untrashing are properly handled, ensuring consistency between WooCommerce and Dokan order statuses. The implementation follows WooCommerce's approach for trash metadata handling and includes error logging for better debugging.

### Feature Video (optional)
N/A

### PR Self Review Checklist:
* [x] Code follows style guidelines
* [x] Naming is clear and self-explanatory
* [x] Code is as simple as possible while meeting requirements
* [x] No unnecessary code repetition
* [x] Code is readable and well-structured
* [x] Performance considerations have been taken into account
* [x] Complex constructions are explained with comments
* [x] No grammar errors in comments and logs

### FOR PR REVIEWER ONLY:
* [ ] Correct — Does the change do what it's supposed to?
* [ ] Secure — Are all inputs properly sanitized and escaped?
* [ ] Readable — Is the code easy to understand?
* [ ] Elegant — Does the change fit within the overall style and architecture?

## Summary by CodeRabbit

- **New Features**
	- Introduced an event listener for managing WooCommerce order actions, specifically for when orders are trashed or restored.
	- Enhanced admin functionality by integrating the new event listener into the existing order management system.

- **Bug Fixes**
	- Improved order status handling in the database to ensure accurate updates when orders are trashed or untrashed.

- **Tests**
	- Added a comprehensive suite of unit tests for the new event listener to validate order trashing and restoring functionalities.

### CODE REVIEW

File: includes/Order/OrderEventListener.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    public function after_order_untrash( int $order_id ) {
+        global $wpdb;
+
+        $order = wc_get_order( $order_id );
+        if ( ! $order instanceof WC_Order ) {
+            dokan_log( "Failed to fetch order {$order_id} after untrash." );
+            return;
+        }
+
+        $previous_status = $order->get_meta( '_wp_trash_meta_status' );
```
Comment (mrabbani) [inline includes/Order/OrderEventListener.php:65]: Is it possible to get status from the order?

```php
$order_status = $order->get_status( 'edit' );
```

File: tests/php/src/Order/OrderEventListenerTest.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    private $seller_id;
+
+    public function setUp(): void {
+        parent::setUp();
+        $this->order_event_listener = new OrderEventListener();
+        $this->seller_id = $this->factory()->seller->create();
+    }
+
+    public function test_processing_order_with_trash_event() {
+        $order_id = $this->create_order( 'processing' );
```
Comment (mrabbani) [inline tests/php/src/Order/OrderEventListenerTest.php:25]: Why are you creating order manually where we have helper functions for creating single & multi-vendor order?

File: tests/php/src/Order/OrderEventListenerTest.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        $vendor_id = dokan_get_seller_id_by_order( $order_id );
+        $initial_balance = dokan()->vendor->get( $vendor_id )->get_earnings( false );
+
+        $processing_order = wc_get_order( $order_id );
+        $processing_order->delete();
+
+        $this->order_event_listener->after_order_trash( $order_id );
```
Comment (mrabbani) [inline tests/php/src/Order/OrderEventListenerTest.php:32]: You should not manually trigger your implementation. It should automatically be called.

Pls follow this instruction for others.

File: tests/php/src/Order/OrderEventListenerTest.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    public function test_completed_order_with_trash_event() {
+        $order_id = $this->create_order( 'completed' );
+        $vendor_id = dokan_get_seller_id_by_order( $order_id );
+        $initial_balance = dokan()->vendor->get( $vendor_id )->get_earnings( false );
+
+        $completed_order = wc_get_order( $order_id );
+        $completed_order->delete();
+
+        $this->order_event_listener->after_order_trash( $order_id );
```
Comment (mrabbani) [inline tests/php/src/Order/OrderEventListenerTest.php:49]: Pls do not call the implementation manually.

**Review by mrabbani — CHANGES_REQUESTED** [review]

File: includes/Order/OrderEventListener.php (hunk 1 above)
Comment (mralaminahamed) [reply includes/Order/OrderEventListener.php:65]: no,  previous status is store in the another meta key. 
Ref: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php#L2360-L2399

File: includes/Order/Controller.php [hunk 5]
```diff
@@ -39,8 +39,9 @@ public function init_classes() {
         $this->container['frontend_hooks'] = new Frontend\Hooks();
 
         if ( is_admin() ) {
-            $this->container['permission']  = new Admin\Permissions();
-            $this->container['admin_hooks'] = new Admin\Hooks();
+            $this->container['permission']     = new Admin\Permissions();
+            $this->container['admin_hooks']    = new Admin\Hooks();
+            $this->container['event_listener'] = new OrderEventListener();
```
[the remark this answers is not in the document]
Comment (mralaminahamed) [reply includes/Order/Controller.php:44]: ok

File: tests/php/src/Order/OrderEventListenerTest.php (hunk 3 above)
Comment (mralaminahamed) [reply tests/php/src/Order/OrderEventListenerTest.php:32]: ok

File: tests/php/src/Order/OrderEventListenerTest.php (hunk 4 above)
Comment (mralaminahamed) [reply tests/php/src/Order/OrderEventListenerTest.php:49]: ok

File: tests/php/src/Order/OrderEventListenerTest.php (hunk 2 above)
Comment (mralaminahamed) [reply tests/php/src/Order/OrderEventListenerTest.php:25]: I created a helper method to make multiple orders without repeating code.

**Review by mrabbani — APPROVED** [review]

**Comment by StalinDurjo** [comment]
If vendor refunds an order, the refunded amount gets subtracted from the withdraw balance resulting in a negative amount in the withdraw balance.

Please refer to the following video for full scenario: https://jam.dev/c/1a35589c-40c5-4039-b3dd-1708c9facf09

@mrabbani bhai @Mohaiminulislam1989 bhai, decision is pending. I didn't receive any decision regarding this issue.

**Comment by mrabbani** [comment]
> If vendor refunds an order, the refunded amount gets subtracted from the withdraw balance resulting in a negative amount in the withdraw balance.
> 
> Please refer to the following video for full scenario: https://jam.dev/c/1a35589c-40c5-4039-b3dd-1708c9facf09
> 
> @mrabbani bhai @Mohaiminulislam1989 bhai, decision is pending. I didn't receive any decision regarding this issue.

We Need further analysis to resolve this issue. For now, We can proceed for release. We will created another issue if needs.

**Pull request merged**